src/neo/SmartContract/Native/NameService.cs

```diff
@@ -169,6 +171,7 @@ private void SetRecord(ApplicationEngine engine, string name, RecordType type, s
             switch (type)
             {
                 case RecordType.A:
+                    if (!ipv4Regex.IsMatch(data)) throw new FormatException();
                     if (!IPAddress.TryParse(data, out IPAddress address)) throw new FormatException();
```

Comment:
it's not checked in TryParse?

Reply:
No, ip string like"0" or "257.0" can pass the check.